proposals/XXXX-compact.md

```diff
@@ -96,79 +104,167 @@ Compact witness type signatures are a kind of foreign function interface (FFI).
 Witness implementations are intended to be provided in some general purpose programming language, which can be called from Compact programs.
 
 An off-chain implementation of a Compact contract consists of the contract's implementation itself, along with some implementation of the witnesses that it uses.
-The mapping from witnesses' Compact type signatures to foreign language type signatures will be predictable and well specified for each supported foreign language.
+The mapping from witnesses' Compact type signatures to foreign language type signatures is predictable and will be well specified for each supported foreign language.
 
 In the specific instantiation of Compact for the Midnight Network, witnesses have TypeScript foreign type signatures and they can be implemented in either TypeScript or JavaScript.
-However, the intention is that other foreign langauges can be equally supported.
+However, the intention is that other foreign languages can be equally well supported.
 
 ## Ledger ADTs
-The public state of a Compact contract is its so-called "ledger".
+The public on-chain state of a Compact contract is its so-called "ledger".
 Compact supports a number of builtin abstract data types (ADTs) for the ledger.
 Examples include cells containing a Compact value, integer counters, sets of Compact values, maps whose keys and values are Compact values, etc.
 
-The specific instantiation of Compact for the Midnight Network has specific ledger ADTs.
+The specific instantiation of Compact for the Midnight Network has specific builtin ledger ADTs.
```

Comment:
Ledger ADTs are defined separately from the compiler; thus the set of ledger ADTs is easily customized without knowledge of or changes to the compiler.